src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const RESULT = [];
+  let newObj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newObj = Object.assign(newObj, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newObj[key];
+        }
+        break;
+
+      case 'clear':
+        newObj = {};
+    }
+    RESULT.push({ ...newObj });
```

Comment:
It is better to push a copy of the object to the array at the end of each loop cycle, but outside of the `switch` block, after the current action is processed. This is already done correctly here, so no action is needed.